### PR TITLE
CI(eos_cli_config_gen): Prepare CI and doc for datamodel refactoring

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Prepare CI and doc for datamodel refactoring

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Add copy of README.md where the new datamodels can be described (list of dicts instead of wildcard keyed dicts)
Add copy of molecule scenario to update test variables to list of dicts.

## How to test

no changes to existing files

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+## Change Summary
```
Comment (carlbuchmann) [inline .github/pull_request_template.md]: This appears to be a duplicate of `PULL_REQUEST_TEMPLATE\pull_request_template.md`

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Review by manuwelakanade — APPROVED** [review]
LGTM!

File: .github/pull_request_template.md (hunk 1 above)
Comment (ClausHolbechArista) [reply .github/pull_request_template.md]: I am investigating this one.

Comment (ClausHolbechArista) [reply .github/pull_request_template.md]: I have removed the copy under the folder. I tested both options, and default template _must_ reside outside the folder, but optional templates must reside _inside_ the folder. The github docs are not completely clear on this.

**Pull request merged**